Framework/Core/src/DataProcessingDevice.cxx

```diff
@@ -239,6 +239,9 @@ bool DataProcessingDevice::ConditionalRun()
     if (info.state != InputChannelState::Running) {
       continue;
     }
+    if (mState.quitRequested) {
```

Comment:
This change is what breaking RootTreeWriterWorkflow: it relies on sink1 to keep forwarding messages from the source to sink2, which sink1 is not doing anymore as it skips incoming inputs entirely. This means we need to rework the enofstream/quit mechanism...